packages/extension-webgazer/src/index.ts

```diff
@@ -0,0 +1,310 @@
+import { JsPsych, JsPsychExtension, JsPsychExtensionParameters } from "jspsych";
+
+// we have to add webgazer to the global window object because webgazer attaches itself to
+// the window when it loads
+declare global {
+  interface Window { webgazer: any; }
+}
+
+interface InitializeParameters extends JsPsychExtensionParameters {
+  /**
+   * Whether to round WebGazer's predicted x, y coordinates to the nearest integer. Recommended
+   * to leave this as `true` because it saves significant space in the data object and the
+   * predictions aren't precise to the level of partial pixels.
+   * @default true
+   */
+  round_predictions: boolean;
+  /**
+   * Whether to initialize WebGazer automatically when the plugin loads. Leave this as `false`
+   * if you plan to initialize WebGazer later in the experiment using a plugin.
+   * @default false
+   */
+  auto_initialize: boolean;
+  /**
+   * The number of milliseconds between each sample. Note that this is only a request, and the 
+   * actual interval will vary depending on processing time.
+   * @default 34
+   */
+  sampling_interval: number;
+  /**
+   * An instance of WebGazer. If left undefined then the global window.webgazer object will be used
+   * if it exists.
+   */
+  webgazer: any;
+}
+
+interface OnStartParameters extends JsPsychExtensionParameters {
+  targets: Array<string>;
+}
+
+class WebGazerExtension implements JsPsychExtension {
+
+  constructor(private jsPsych: JsPsych) { }
+
+  // private state for the extension
+  // extension authors can define public functions to interact
+  // with the state. recommend not exposing state directly
+  // so that state manipulations are checked.
+  private currentTrialData = [];
+  private currentTrialTargets = {};
+  private currentTrialSelectors:Array<string>;
+  private domObserver:MutationObserver;
+  private webgazer;
+  private initialized:boolean = false;
+  private currentTrialStart:number;
+  private activeTrial:boolean = false;
```

Comment:
Great! In these cases, TS infers the boolean type automatically (type widening):
```suggestion
  private initialized = false;
  private currentTrialStart:number;
  private activeTrial = false;
```